monero/backends/jsonrpc/daemon.py

```diff
@@ -587,68 +587,85 @@ def hard_fork_info(self):
 
         return self.raw_jsonrpc_request('hard_fork_info')
 
-    def set_bans(self, bans):
+    def set_bans(self, ip, ban, seconds):
         """
-        Ban (or uban) other nodes by IP for a certain length of time.
+        Ban (or unban) other nodes by IP for a certain length of time.
 
-        :param bans: list of dict ban entries with the following structure:
-
-        ban = {
-            "host": str; Host to ban (IP in A.B.C.D form - will support I2P address in the future),
-            "ip": unsigned int; IP address to ban, in int format,
-            "ban": bool; Set true to ban,
-            "seconds": unsigned int; Number of seconds to ban node
-        }
+        :param ip: ip address of node to ban in two forms: 'a.b.c.d' str or 32-bit unsigned integer
+        :param bool ban: Whether to ban (True) or unban (False)
+        :param int seconds: Number of seconds to ban
 
         Notes:
-        * For a given entry, only one of "host" or "ip" must be provided, but both may be provided so long as they match.
-        * "seconds" does not need to be provided if "ban" is False
+        * Params "ip", "ban", and "seconds" can be single elements or they can be iterables of the same length. For example:
+
+            ips =        ['229.52.8.67', 611282986]
+            should_ban = [True,          False]
+            seconds =    [10000,         None]
+            daemon.set_bans(ips, should_ban, seconds)
+
+        * "seconds" can be 0 or None if "ban" is False
 
         Output:
         {
         "status": str; General RPC error code. "OK" means everything looks good. Any other value means that something went wrong.
         }
         """
 
-        if isinstance(bans, dict):
-            bans = [bans]
-
-        # Validate bans
-        for ban_info in bans:
-            host = ban_info.get('host')
-            ip = ban_info.get('ip')
-            ban = ban_info.get('ban')
-            seconds = ban_info.get('seconds')
-            if host is None and ip is None:
-                raise ValueError('host or ip must be provided for each ban')
-            elif ban is None:
-                raise ValueError('"ban" value must be provided for each ban entry')
-            elif seconds is None and ban:
-                raise ValueError('"seconds" value must be provided when an IP is banned')
-            if host:
-                if len(host.split('.')) != 4:
-                    raise ValueError('host "{}" is incorrectly formatted'.format(host))
-                for ip_part in host.split('.'):
+        def is_valid_ip(ip):
```

Comment:
I thought about using the ipaddress module, but I didn't because it isn't standard in < Python 3.3, and I didn't want to add a dependency unless necessary. It would be easy enough to add it though, and would definitely make the code cleaner. It's up to you.

Reply:
I think additional dependency for Python2 won't do any harm. It seems a compatible module is available: https://pypi.org/project/py2-ipaddress/

I've already split `test_requirements*.txt` between Py2 and Py3. The same method could be used to make separate `requirements.txt`.

Reply:
We might not even need to split it (yet). [This pip package](https://pypi.org/project/ipaddress/) is the official package I'm pretty sure, and it works for all relevant Python versions.